### PR TITLE
Add recommended add-ons to the setup wizard

### LINKED ISSUE #1659: Add JS Scripting installation for Blockly in script creation & in wizard as well as rrd4j
With Blockly support for JS Scripting merged in #1617 and the NashornJS engine removed from core, Blockly depends on at least one JS automation addon being installed. For several reasons, the GraalJS based JS Scripting add-on is the way to take for the future.

Installing JS Scripting (GraalJS) by default was rejected in https://github.com/openhab/openhab-distro/pull/1455 with some good reasons, the consensus was to handle JS Scripting installation through the first start wizard and possibly the script creation page where you can select Blockly as an option as well.

Discussed in this distro PR as well: addind rrd4j persistence to the installtion wizard and not installing it by default.

/cc @rkoshak @stefan-hoehn @ghys

### PR DESCRIPTION
**Pull request by florian-h05**
Closes #1659.

This adds the ability to define a list of recommended add-ons which will be pre-selected in the setup wizard.

Currently, JS Scripting (GraalJS) Automation as a Blockly Dependency and RRD4J Persistence as default persistence service (used e.g. for trend-line, analyzer in the UI) are selected as recommended add-ons.

A note about recommended add-ons in general is displayed.

### CODE REVIEW

File: bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
   "setupwizard.addons.title": "Install Add-ons",
   "setupwizard.addons.header1": "Most of openHAB's functionality is provided by add-ons.",
   "setupwizard.addons.header2": "Choose which ones you'd like to install right away.",
+  "setupwizard.addons.header3": "Add-ons recommended for functional reasons are pre-selected.",
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json:24]: @rkoshak What do you think about the wording?

File: bundles/org.openhab.ui/web/src/js/routes.js [hunk 2]
```diff
@@ -102,6 +102,7 @@ export default [
   },
   {
     path: '/setup-wizard/',
+    beforeEnter: [enforceAdminForRoute],
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/js/routes.js:105]: I have tested this on a fresh, clean openHAB installation, and you have to create an admin account before entering the wizard so this does not cause problems.

File: bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json (hunk 1 above)
Comment (rkoshak) [reply bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json:24]: As written it seems mysterious and raises questions. What functional reasons? Maybe something like

> We've pre-selected some add-ons to provide a fully capable environment to start with.

I'm not really happy with that wording either but it at least heads off the questions raised.

Comment (digitaldan) [reply bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json:24]: I would keep the language plain, how about:
```
"setupwizard.addons.title": "Install Add-ons",
"setupwizard.addons.header1": "Add-ons extend openHAB's functionality.",
"setupwizard.addons.header2": "Choose which add-ons you would like to install now.",
"setupwizard.addons.header3": "A recommended set of add-ons have been pre-selected.",
```

Comment (rkoshak) [reply bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json:24]: Even better. I like that wording the best!

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json:24]: Okay, thanks for your input.
Much better than what I came up with!

I’ll change it.

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/assets/i18n/setup-wizard/en.json]: Done, I have changed 2 and 3 and left 1 as is. I prefer the current wording for 1.

**Review by ghys — APPROVED** [review]
This looks good, thanks!

**Pull request merged**